_gen/gen.go

```diff
@@ -413,7 +454,10 @@ func genMulAvx2(name string, inputs int, outputs int, xor bool) {
 	for _, ptr := range inPtrs {
 		ADDQ(offset, ptr)
 	}
-	// Offset no longer needed unless not regdst
+	// Offset no longer needed unless not regDst
+	if !regDst {
+		SHRQ(U8(3), offset) // divide by 8 since we'll be scaling it up when loading or storing
+	}
```

Comment:
Can we avoid this? Some Intel react badly to scaling in address resolution, so I tend to avoid it if it is feasible.